src/backend/base/langflow/graph/graph/base.py

```diff
@@ -195,33 +195,29 @@ def add_nodes_and_edges(self, nodes: list[NodeData], edges: list[EdgeData]):
         self._edges = self._graph_data["edges"]
         self.initialize()
 
-    def add_component(self, _id: str, component: "Component"):
-        if _id in self.vertex_map:
-            return
+    def add_component(self, component: "Component", component_id: Optional[str] = None) -> str:
+        component_id = component_id or str(component.name + "-" + str(uuid.uuid4()))
+        if component_id in self.vertex_map:
+            raise ValueError(f"Component ID {component_id} already exists")
+        if not component_id.startswith(component.name):
+            raise ValueError(f"Component ID {component_id} does not match component name {component.name}")
         frontend_node = component.to_frontend_node()
-        frontend_node["data"]["id"] = _id
-        frontend_node["id"] = _id
+        frontend_node["data"]["id"] = component_id
+        frontend_node["id"] = component_id
         self._vertices.append(frontend_node)
         vertex = self._create_vertex(frontend_node)
         vertex.add_component_instance(component)
         self.vertices.append(vertex)
-        self.vertex_map[_id] = vertex
-
-        if component._edges:
-            for edge in component._edges:
-                self._add_edge(edge)
-
-        if component._components:
-            for _component in component._components:
-                self.add_component(_component._id, _component)
+        self.vertex_map[component_id] = vertex
+        return component_id
 
```

Comment:
I don't know the graph code well enough, but this seems like a big change. Can you explain what the bug was here and what this change fixes?